debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -304,6 +306,34 @@ public Long lastReceivedLsn() {
                 return lastReceivedLsn != null ? lastReceivedLsn.asLong() : null;
             }
 
+            @Override
+            public void startKeepAlive(ExecutorService service) {
+                if (keepAliveExecutor == null) {
+                    keepAliveExecutor = service;
+                    keepAliveExecutor.submit(() -> {
+                        while (!Thread.currentThread().isInterrupted()) {
```

Comment:
Reading the docs of `shutdownNow()` it seems it's not guaranteed that it actually will interrupt the running thread. I think a safer way would be to have a `AtomicBoolean` or so as an instance field in this class and use this as the stopping signal.